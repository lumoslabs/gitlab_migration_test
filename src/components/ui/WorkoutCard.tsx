import React from 'react'
import { Card } from 'react-bootstrap'
import { css, StyleSheet } from 'aphrodite/no-important'
import commonStyles from '@styles/commonStyles'
import { base } from '@styles/colors'
import WideActionButton from '@components/ui/WideActionButton'

export interface IWorkoutCardProps {
  clickHandler(e: React.MouseEvent<any>): any;
  remainingGamesCount: number,
  totalGameCount: number
}

const WorkoutCard = ({ clickHandler, remainingGamesCount = 3, totalGameCount = 3 }: IWorkoutCardProps): JSX.Element => {
  // Workout Card Icon
  let workoutIconUrl = '/assets/workout_icon.svg'
  if (remainingGamesCount === 0) {
    workoutIconUrl = '/assets/training_complete.svg'
  }
  if (remainingGamesCount < totalGameCount && remainingGamesCount > 0) {
    workoutIconUrl = `/assets/training_${totalGameCount - remainingGamesCount}by3.svg`
  }

  // Workout Card Button Text
  let buttonText = 'Start My Workout'
  if (remainingGamesCount < totalGameCount) {
    buttonText = 'Resume Workout'
  }
  if (remainingGamesCount === 0) {
    buttonText = 'Workout Complete'
  }

  return (
    <div className={css([commonStyles.flexColumnAllCenter, styles.section])}>
      <Card className={css([commonStyles.flexColumnAllCenter, styles.card])}
        onClick={clickHandler}
      >
        <Card.Body className={css([commonStyles.flexColumnAllCenter, styles.cardBody])}>
          <Card.Img className={css(styles.cardImg)}
            src={workoutIconUrl}
            alt='Workout Icon'
          >
          </Card.Img>
          <Card.Title className={css(styles.cardTitle)}>
            {'Daily Workout'}
          </Card.Title>
          <Card.Subtitle className={css(styles.cardSubtitle)}>
            {'Your daily workout of 3 games.'}
          </Card.Subtitle>
          <WideActionButton
            disabled={remainingGamesCount === 0}
            extendStyles={styles.cta}
            buttonText={buttonText || 'Start My Workout'}
            onClick={clickHandler}
            eventData={{ id: 'start_workout' }}
          />
        </Card.Body>
      </Card>
    </div>
  )
}

const styles = StyleSheet.create({
  card: {
    borderRadius: '4vmin',
    border: 'none',
    boxShadow: `0px 0.125vmin 0.125vmin 0px rgba(0, 0, 0, 0.11),
      0px 0.25vmin 0.5vmin 0px rgba(0, 0, 0, 0.13),
      0px 0.25vmin 0.75vmin 0px rgba(0, 0, 0, 0.1)`
  },

  section: {
    marginRight: '2vmin'
  },

  cardTitle: {
    fontSize: '32x',
    fontFamily: 'MuseoSans700',
    fontWeight: 700,
    lineHeight: '4.5vmin'
  },

  cardSubtitle: {
    color: base.gray8,
    fontSize: '2.75vmin',
    fontFamily: 'MuseoSans500',
    fontWeight: 500,
    lineHeight: '3.25vmin'
  },

  cardImg: {
    width: '19.5vmin',
    height: '19.5vmin'
  },

  cardBody: {
    width: '30vmax',
    height: '60vmin',
    justifyContent: 'space-evenly'
  },

  cta: {
    height: '8.75vmin',
    width: '42.625vmin',
    borderRadius: '4.375vmin',
    fontFamily: 'MuseoSans700',
    fontSize: '3.625vmin',
    fontWeight: 700
  }
})

export default WorkoutCard
