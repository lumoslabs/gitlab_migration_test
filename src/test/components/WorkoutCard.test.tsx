import React from 'react'
import WorkoutCard from '@components/ui/WorkoutCard'
import { render } from '@testing-library/react'

const handleClick = () => { window.location.href = '/' }

it('matches snapshot', () => {
  const { asFragment } = render(
    <WorkoutCard clickHandler={handleClick} remainingGamesCount={3} totalGameCount={3} />, {}
  )
  expect(asFragment()).toMatchSnapshot()
})
