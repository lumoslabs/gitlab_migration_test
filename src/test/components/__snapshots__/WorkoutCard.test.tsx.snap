// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flexColumnAllCenter_1vexgx1-o_O-section_rgclyh"
  >
    <div
      class="flexColumnAllCenter_1vexgx1-o_O-card_1rrn0ox card"
    >
      <div
        class="flexColumnAllCenter_1vexgx1-o_O-cardBody_1858xbs card-body"
      >
        <img
          alt="Workout Icon"
          class="card-img cardImg_3mv33q"
          src="/assets/workout_icon.svg"
        />
        <div
          class="cardTitle_ena3cq card-title h5"
        >
          Daily Workout
        </div>
        <div
          class="cardSubtitle_1g4oaq4 card-subtitle h6"
        >
          Your daily workout of 3 games.
        </div>
        <div
          class="flexJustifyCenter_jfsfb5"
        >
          <button
            class="button_1xm70sk-o_O-wideAction_4zoj1f-o_O-cta_1jbl92e"
          >
            Start My Workout
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
